Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,648 @@
+:navigation-title: List of Validators
+
+..  include:: /Includes.rst.txt
+..  _extbase-validator:
+
+=======================================
+Built-in validators provided by Extbase
+=======================================
+
+This document lists all built-in Extbase validators,
+along with example usage for each using PHP attributes.
+
+..  seealso::
+    *   `Using validation for Extbase models and controllers <https://docs.typo3.org/permalink/t3coreapi:extbase-validation>`_
+    *   `Custom Extbase validator implementation <https://docs.typo3.org/permalink/t3coreapi:extbase-domain-validator>`_
+
+..  note::
+    All validators except the `NotEmptyValidator <https://docs.typo3.org/permalink/t3coreapi:extbase-validator-notempty>`_
+    accept empty values as valid. If empty values should not be possible,
+    combine these validators with the `NotEmptyValidator`.
+
+..  contents:: Validators in Extbase
+
+..  _extbase-validator-alphanumeric:
+
+AlphanumericValidator
+=====================
+
+Checks that a value contains **only letters and numbers** — no spaces, symbols,
+or special characters.
+
+This includes letters from **many languages**, not just A–Z. For example,
+letters from alphabets like **Hebrew, Arabic, Cyrillic**, and others are also
+allowed.
+
+This is useful for fields like usernames or codes where only plain text
+characters are allowed.
+
+If you want to allow any symbols (like `@`, `#`, `-`) or spaces, use the
+`RegularExpressionValidator <https://docs.typo3.org/permalink/t3coreapi:extbase-validator-regularexpression>`_
+instead.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Alphanumeric'])]
+    protected string $username;
+
+..  _extbase-validator-boolean:
+
+BooleanValidator
+================
+
+Checks if a value matches a specific boolean value (`true` or `false`).
+
+This validator is useful when you want to enforce that a property explicitly
+evaluates to either `true` or `false`, such as for checkboxes in forms.
+
+By default, it accepts any boolean value unless the `is` option is
+set.
+
+Options:
+
+`is`
```

Comment:
The main purpose of this option is not mentioned: requiring exactly the boolean value set by `is`.

Funny that Extbase does not have a basic `EqualValidator` which would cover this generally for all validators but that's how it is.

Reply:
Checks if a value matches a specific boolean value (`true` or `false`). sagt das meiner meinung nach aus?

Reply:
Yup, sounds good. :-)